apps/web/src/components/WalletConnection/ConnectWalletButton.tsx

```diff
@@ -1,12 +1,34 @@
 /* eslint-disable */
-import React, { useState } from 'react'
+import React, { useEffect, useState } from 'react'
 import { useWeb3React } from '@web3-react/core'
 import { InjectedConnector } from '@web3-react/injected-connector'
 import { BoxLayout, Text, StackLayout } from '@bloxifi/ui'
 
+const supportedChains = {
+  1: 'mainnet',
+  3: 'ropsten',
+  4: 'rinkeby',
+  42: 'kovan',
+  1284: 'moonbeam',
+}
+
+declare global {
+  interface Window {
+    ethereum?: any
+  }
+}
+
 export const ConnectWalletButton = () => {
   const [hasError, setHasError] = useState(false)
+  const [currentNetwork, setCurrentNetwork] = useState<number | undefined>()
   const web3Context = useWeb3React()
+  const isWrongNetworkConnected =
+    currentNetwork && !supportedChains[currentNetwork]
+  const isMetamaskInstalled = typeof window.ethereum !== 'undefined'
```

Comment:
we should use https://www.npmjs.com/package/@metamask/detect-provider
with option `mustBeMetamask` because that's all we need. I've seen that `window.ethereum` doesn't mean that we have Metamask installed as per this documentation https://docs.metamask.io/guide/ethereum-provider.html

Reply:
also, we should add this info in UserContainer, because this is some kind of authentication.